src/ModelStates/Transition.php

```diff
@@ -21,6 +21,26 @@ public function canTransition(): bool
         return true;
     }
 
+    public function direction(): TransitionDirection
+    {
+        return TransitionDirection::FORWARD();
+    }
+
+    public function isVisibleFromUserFrontend(): bool
+    {
+        return true;
+    }
+
+    public function to(): State
+    {
+        return new $this->to();
```

Comment:
```suggestion
        return new $this->to;
```

or

```suggestion
        return (new $this->to)();
```

Reply:
`$this->to` is a `class-string<State> $to`. So to me, the instantiation looks fine... It also works fine as we can see in the test.

`self::assertSame(StateX::name(), $last->from()::name());`

Reply:
I just find the operator precedence really unexpected here, at first glance I though this is a recursive call.